samples/unit-tests/stock-tools/indicators/demo.js

```diff
@@ -42,89 +42,605 @@ QUnit.test('Managing tech indicators in Stock Tools', function (assert) {
+    );
+    assert.close(
+        parseFloat(chart.yAxis[0].options.top),
+        0,
+        0.0001, // up to 0.0001% is fine
+        'Correct Top for Main axis after adding 2 indicators. Main axis originally 100% height'
+    );
+    assert.close(
+        parseFloat(chart.yAxis[2].options.height),
+        20 / 1.4,
+        0.0001, // up to 0.0001% is fine
+        'Correct Height for first indicator axis after adding 2 indicators. Main axis originally 100% height'
+    );
+    assert.close(
+        parseFloat(chart.yAxis[2].options.top),
+        100 / 1.4,
+        0.0001, // up to 0.0001% is fine
+        'Correct Top for first indicator axis after adding 2 indicators. Main axis originally 100% height'
+    );
+
+    assert.close(
+        parseFloat(chart.yAxis[3].options.height),
+        20 / 1.4,
+        0.0001, // up to 0.0001% is fine
+        'Correct Height for second indicator axis after adding 2 indicators. Main axis originally 100% height'
+    );
+    assert.close(
+        parseFloat(chart.yAxis[3].options.top),
+        120 / 1.4,
+        0.0001, // up to 0.0001% is fine
+        'Correct Top for second indicator axis after adding 2 indicators. Main axis originally 100% height'
+    );
+
+    chart.navigationBindings.utils.manageIndicators.call(
+        chart.navigationBindings,
+        {
+            actionType: 'remove',
+            seriesId:
+                chart.series[chart.series.length - 2]
+                    .options.id
+        }
+    );
+
+    assert.close(
+        parseFloat(chart.yAxis[0].options.height),
+        100 / 1.2,
+        0.0001, // up to 0.0001% is fine
+        'Correct Height for Main axis after deleting 1 indicator. Main axis -> 100% height'
+    );
+
+    assert.close(
+        parseFloat(chart.yAxis[0].options.top),
+        0,
+        0.0001, // up to 0.0001% is fine
+        'Correct top for Main axis after deleting 1 indicator. Main axis -> 100% height'
+    );
+
+    assert.close(
+        parseFloat(chart.yAxis[2].options.height),
+        20 / 1.2,
+        0.0001, // up to 0.0001% is fine
+        'Correct height for indicator axis after deleting 1 indicator. Main axis -> 100% height'
+    );
+
+    assert.close(
+        parseFloat(chart.yAxis[2].options.top),
+        100 / 1.2,
+        0.0001, // up to 0.0001% is fine
+        'Correct top for indicator axis after deleting 1 indicator. Main axis -> 100% height'
+    );
+
+    chart.navigationBindings.utils.manageIndicators.call(
+        chart.navigationBindings,
+        {
+            actionType: 'remove',
+            seriesId:
+                chart.series[chart.series.length - 2]
+                    .options.id
+        }
+    );
+
+    assert.close(
+        parseFloat(chart.yAxis[0].options.height),
+        100,
+        0.0001, // up to 0.0001% is fine
+        'Correct Height for Main axis after deleting all indicators. Main axis -> 100% height'
+    );
+
+    chart.destroy();
```

Comment:
Is it necessary to recreate the chart?

Reply:
I replaced the recreating of the chart with updates.